### DISCUSSION TITLE
What does 'isolate' and 'extend' mean?

### DISCUSSION THREAD

**Discussion by UchihaYuki** [ocombe/ng2-translate#1507] [Q&A] [answered]
I'm still confused after reading the doc. Please give a simple example for each.
reference: https://ngx-translate.org/reference/configuration/

Edit:
What's the relationship between `translate.translations` and `translate.store.translations`? What's the relationship between them and, 'isolate' 'and extend'?

**Comment by UchihaYuki** [comment]
It seems non-isolated translations will be put into `translate.store.translations`. But I still have no clue what will 'extend' do.

**Comment by UchihaYuki** [comment]
After some testing, I think 'extend' means to extend the `translations `in `translateStore.translations`.

**Comment by rbalet** [comment, marked as answer]
https://ngx-translate.org/reference/ngmodules/#configuration-options


Name | Type | Description
-- | -- | --
extend | boolean | Default: false. Extends translations for a given language instead of replacing them.
isolate | boolean | Default: false. Isolates the service instance - making allows independent switching of languages.
